lib/floe/workflow/states/mapped_state.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Floe
+  class Workflow
+    module States
+      class MappedState < Floe::Workflow::State
+        attr_reader :input_path, :output_path
+
+        def initialize(workflow, name, payload)
+          super
+          @input_path  = Path.new(payload.fetch("InputPath", "$"))
+          @output_path = Path.new(payload.fetch("OutputPath", "$"))
```

Comment:
Larger question.

How do we want to handle invalid State payloads?

Do we want to throw `IllegalArgument` exceptions when the states are invalid?